### ISSUE TITLE
wen logits

### ISSUE THREAD

**Issue by steffencruz** [macrocosm-os/prompting#405]
Multiple choice task should be based on logits. This improves the task in two ways:
- Avoid brittle parsing logic when extracting miner prediction from text string
- Measure confidence of miner response

Miners should respond with a string that we can load as JSON. This will enable us to score the (re-normalized) probability assigned to the correct choice instead of a binary reward signal.

## Example
```python
miner_response = {
    "A": 1,
    "B": 0.6,
    "C": 0.25,
    "D": 0.05
}
```
Proposed reward mechanism:
```python
reward = miner_response.get( correct_response, 0 )
```
So if the  correct answer is `"C"` the miner will score `0.25` instead of `0`, which is what the current mechanism would reward.

Note that the current approach is equivalent to `miner_reward = ( argmax(miner_response) == true_index )`.
